### PR TITLE
Jihad C4 (for real this time!)

### PR DESCRIPTION
**Pull request by P250**
An absolute completed system for the Jihad C4, that hopefully manages many of the edge cases! Here is the feature list: 

- Round start gives a random Terrorist the jihad c4
- Ability to give, pass on, and use the jihad c4
- EmitSound support (world sounds, not client-sided ones)
- Jihad c4 explosion physically pushes players slightly and damages them according to the old formula
- Notifications support

Latest version of CS# (v233) and latest version of metamod

The EmitSound support uses a function signature but it is a very common one (from cs2fixes) and if it changes it will surely be updated by them immediately again. 

A consequence of this system means multiple players can have a jihad c4 at the same time! <- NOT TESTED.

I have left a debug command in there (css_d) for easy testing of the jihad c4, but it should be removed of course!

### CODE REVIEW

**Comment by MSWS** [comment]
Is there a way to remove the added references to the CSS API? Ideally, upgrading the CSS API version should only require one change of the number, and it shouldn't be referenced in other submodules. (We've been developing without referencing it, so I assume it's possible).

**Comment by P250** [comment]
> Is there a way to remove the added references to the CSS API? Ideally, upgrading the CSS API version should only require one change of the number, and it shouldn't be referenced in other submodules. (We've been developing without referencing it, so I assume it's possible).

For my workspace if I remove the added references I get some errors, but I could remove it anyway and commit that if you'd like?

File: mod/Jailbreak.Rebel/JihadC4/JihadC4Behavior.cs [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    private IJihadC4Notifications _jihadNotifications;
+    private BasePlugin? _basePlugin;
+
+    // EmitSound(CBaseEntity* pEnt, const char* sSoundName, int nPitch, float flVolume, float flDelay)
+    private readonly MemoryFunctionVoid<CBaseEntity, string, int, float, float> CBaseEntity_EmitSoundParamsLinux; // LINUX ONLY.
+
+    public JihadC4Behavior(IJihadC4Notifications jihadC4Notifications)
+    {
+        _jihadNotifications = jihadC4Notifications;
+        CBaseEntity_EmitSoundParamsLinux = new("48 B8 ? ? ? ? ? ? ? ? 55 48 89 E5 41 55 41 54 49 89 FC 53 48 89 F3");
```
Comment (jithatsonei) [inline mod/Jailbreak.Rebel/JihadC4/JihadC4Behavior.cs:32]: prefer to see this implemented for the entire gamemode to use (ex warden sounds) but that can wait

File: mod/Jailbreak.Rebel/JihadC4/JihadC4Behavior.cs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            // This will deal with the explosion and ensuring the detonator is killed, as well as removing the bomb entity.
+            TryDetonateJihadC4(player, metadata.Delay, bombEntity); 
+
+            TryEmitSound(player, "jb.jihad", 1, 1f, 0f);
+            _jihadNotifications.PlayerDetonateC4(player).ToAllChat();
+
+        }
+    }
+
+    // Todo please remove later!!
```
Comment (jithatsonei) [inline mod/Jailbreak.Rebel/JihadC4/JihadC4Behavior.cs:82]: :[

File: mod/Jailbreak.Rebel/JihadC4/JihadC4Behavior.cs [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+    [GameEventHandler]
+    public HookResult OnPlayerPickupC4(EventBombPickup @event, GameEventInfo info)
+    {
+        CCSPlayerController? player = @event.Userid;
+        if (player == null || !player.IsValid) { return HookResult.Continue; }
+
+        CPlayer_WeaponServices? weaponServices = player.PlayerPawn?.Value?.WeaponServices;
+        if (weaponServices == null) { return HookResult.Continue; }
+
+        CC4 bombEntity = new CC4(weaponServices.MyWeapons.Last()!.Value!.Handle); // The last item in the weapons list is the last item the player picked up, apparently.
```
Comment (jithatsonei) [inline mod/Jailbreak.Rebel/JihadC4/JihadC4Behavior.cs:129]: my eyes

**Review by jithatsonei — CHANGES_REQUESTED** [review]
fix the :[ comment then LGTM

**Pull request merged**